packages/front-end/pages/datasources/queries/[did].tsx

```diff
@@ -215,6 +254,29 @@ const DataSourceQueries = (): React.ReactElement => {
                   </Tooltip>
                 </td>
                 <td>{query.externalId || "N/A"}</td>
+                <td>
+                  <div className="d-flex align-items-center">
+                    {supportsQueryCancellation &&
+                      canCancelQueries &&
+                      ["queued", "running"].includes(query.status) && (
+                        <FaStopCircle
+                          className="text-danger mr-2"
+                          onClick={(e) => {
+                            e.preventDefault();
+                            e.stopPropagation();
+                            apiCall<Response>(
+                              `/datasource/${did}/cancel/${query.id}`,
+                              {
+                                method: "POST",
+                              }
+                            );
```

Comment:
I could use some input on what to do with the response to this api call. I wasn't sure what the best way to surface an error message or success status to the user would be, as the UI on this page is already getting somewhat cluttered.